components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -237,107 +246,121 @@ public void registerOAuthApplicationData(OAuthConsumerAppDTO application) throws
+                        app.setTokenBindingValidationEnabled(application.isTokenBindingValidationEnabled());
+                        app.setTokenRevocationWithIDPSessionTerminationEnabled(
+                                application.isTokenRevocationWithIDPSessionTerminationEnabled());
                     }
-                    app.setBypassClientCredentials(application.isBypassClientCredentials());
-                    app.setRenewRefreshTokenEnabled(application.getRenewRefreshTokenEnabled());
-                    validateBindingType(application.getTokenBindingType());
-                    app.setTokenBindingType(application.getTokenBindingType());
-                    app.setTokenBindingValidationEnabled(application.isTokenBindingValidationEnabled());
-                    app.setTokenRevocationWithIDPSessionTerminationEnabled(
-                            application.isTokenRevocationWithIDPSessionTerminationEnabled());
-                }
-                dao.addOAuthApplication(app);
-                AppInfoCache.getInstance().addToCache(app.getOauthConsumerKey(), app);
-                if (LOG.isDebugEnabled()) {
-                    LOG.debug("Oauth Application registration success : " + application.getApplicationName() + " in " +
-                            "tenant domain: " + tenantDomain);
+                    dao.addOAuthApplication(app);
+                    AppInfoCache.getInstance().addToCache(app.getOauthConsumerKey(), app);
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Oauth Application registration success : " + application.getApplicationName() +
+                                " in " +
+                                "tenant domain: " + tenantDomain);
+                    }
+                } else {
+                    String message = "No application details in the request. Failed to register OAuth App.";
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(message);
+                    }
+                    throw handleClientError(INVALID_REQUEST, message);
                 }
             } else {
-                String message = "No application details in the request. Failed to register OAuth App.";
                 if (LOG.isDebugEnabled()) {
-                    LOG.debug(message);
-                }
-                throw handleClientError(INVALID_REQUEST, message);
-            }
-        } else {
-            if (LOG.isDebugEnabled()) {
-                if (application != null) {
-                    LOG.debug("No authenticated user found. Failed to register OAuth App: " +
-                            application.getApplicationName());
-                } else {
-                    LOG.debug("No authenticated user found. Failed to register OAuth App");
+                    if (application != null) {
+                        LOG.debug("No authenticated user found. Failed to register OAuth App: " +
```

Comment:
do if(log.isDebugenabled()) check since we do a concat operation in the debug msg. Since it is a costly operation, better to do a check whether debug is enabled and add debug log

Reply:
The debug enabled checked is performed at here.
https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/41ca48815ee5560546692c8864ce0ca68ffe37b4/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java#L351